### PR TITLE
sauce-connect: 4.5.1 -> 4.5.4

### PR DESCRIPTION
**Pull request by Fresheyeball**
###### Motivation for this change

Update the version of sauce-connect needed for websocket support

###### Things done

Update and tested on nixos. Used prefetch to get hash for other platforms.

- [x ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/sauce-connect/default.nix [hunk 1]
```diff
@@ -4,18 +4,18 @@ with lib;
 
 stdenv.mkDerivation rec {
   name = "sauce-connect-${version}";
-  version = "4.5.1";
+  version = "4.5.4";
 
   src = fetchurl (
     if stdenv.hostPlatform.system == "x86_64-linux" then {
```
Comment (jonringer) [inline pkgs/development/tools/sauce-connect/default.nix]: you may want to do an assertion that checks to see if the else branch will land on a darwin system.

```suggestion
    assertOneOf "hostPlatform" stdenv.hostPlatform.system [ "x86_64-linux" "i686-linux" "x86_64-darwin" ]
    if stdenv.hostPlatform.system == "x86_64-linux" then {
```

File: pkgs/development/tools/sauce-connect/default.nix [hunk 2]
```diff
@@ -4,18 +4,18 @@ with lib;
 
 stdenv.mkDerivation rec {
   name = "sauce-connect-${version}";
-  version = "4.5.1";
+  version = "4.5.4";
 
   src = fetchurl (
```
Comment (jonringer) [inline pkgs/development/tools/sauce-connect/default.nix]: maybe bring in the hostPlatform namespace so you can use just "system" instead of that long path?
```suggestion
  src = with stdenv.hostPlatform; fetchurl (
```

**Review by jonringer — COMMENTED** [review]
kind of nit picks?

File: pkgs/development/tools/sauce-connect/default.nix [hunk 3]
```diff
@@ -4,18 +4,19 @@ with lib;
 
 stdenv.mkDerivation rec {
   name = "sauce-connect-${version}";
-  version = "4.5.1";
+  version = "4.5.4";
 
-  src = fetchurl (
+   src = with stdenv.hostPlatform; fetchurl (
+    assertOneOf "hostPlatform" stdenv.hostPlatform.system [ "x86_64-linux" "i686-linux" "x86_64-darwin" ]
```
Comment (jonringer) [inline pkgs/development/tools/sauce-connect/default.nix]:
```suggestion
    assertOneOf "hostPlatform" system [ "x86_64-linux" "i686-linux" "x86_64-darwin" ]
```

File: pkgs/development/tools/sauce-connect/default.nix [hunk 4]
```diff
@@ -4,18 +4,19 @@ with lib;
...
     if stdenv.hostPlatform.system == "x86_64-linux" then {
       url = "https://saucelabs.com/downloads/sc-${version}-linux.tar.gz";
-      sha256 = "0lpfvlax7k8r65bh01i3kzrlmx0vnm9vhhir8k1gp2f4rv6z4lyx";
+      sha256 = "1w8fw47q4bzpk5jfagmc0cbp69jdd6jcv2xl1gx91cbp7xd8mcbf";
     } else if stdenv.hostPlatform.system == "i686-linux" then {
```
Comment (jonringer) [inline pkgs/development/tools/sauce-connect/default.nix]:
```suggestion
    } else if system == "i686-linux" then {
```

**Review by jonringer — COMMENTED** [review]
Now you can shorten this to just using system :)

**Comment by drewrisinger** [comment]
(triage):
* [ ] Needs rebased for merge conflicts
* [ ] Code review comments need applied.
* [ ] tests of some sort need run.
* [ ] Commits need squashed
